src/test/java/com/stripe/functional/CouponTest.java

```diff
@@ -1,128 +1,89 @@
 
-  @Test
-  public void testCustomerCreateWithCoupon() throws StripeException {
-    Coupon coupon = Coupon.create(getUniqueCouponParams());
-    Map<String, Object> customerWithCouponParams = new HashMap<String, Object>();
-    customerWithCouponParams.put("coupon", coupon.getId());
-    Customer customer = Customer.create(customerWithCouponParams);
-    assertEquals(customer.getDiscount().getCoupon().getId(), coupon.getId());
-
-    customer.deleteDiscount();
-    assertNull(Customer.retrieve(customer.getId()).getDiscount());
+    verifyRequest(
+        APIResource.RequestMethod.POST,
+        String.format("/v1/coupons/%s", resource.getId()),
+        params
+    );
   }
 
   @Test
-  public void testCouponCreatePerCallAPIKey() throws StripeException {
-    Coupon coupon = Coupon.create(getUniqueCouponParams(), Stripe.apiKey);
-    assertEquals(coupon.getDuration(), "once");
-  }
+  public void testList() throws StripeException {
+    Map<String, Object> listParams = new HashMap<String, Object>();
+    listParams.put("limit", 1);
 
-  @Test
-  public void testCouponRetrievePerCallAPIKey() throws StripeException {
-    Coupon createdCoupon = Coupon.create(getUniqueCouponParams(),
-        Stripe.apiKey);
-    Coupon retrievedCoupon = Coupon.retrieve(createdCoupon.getId(),
-        Stripe.apiKey);
-    assertEquals(createdCoupon.getId(), retrievedCoupon.getId());
-  }
+    CouponCollection resources = Coupon.list(listParams);
 
-  @Test
-  public void testCouponListPerCallAPIKey() throws StripeException {
-    Map<String, Object> listParams = new HashMap<String, Object>();
-    listParams.put("count", 1);
-    List<Coupon> coupons = Coupon.all(listParams, Stripe.apiKey).getData();
-    assertEquals(coupons.size(), 1);
+    assertNotNull(resources);
+    verifyRequest(
+        APIResource.RequestMethod.GET,
+        String.format("/v1/coupons")
+    );
   }
 
   @Test
-  public void testCouponDeletePerCallAPIKey() throws StripeException {
-    Coupon createdCoupon = Coupon.create(getUniqueCouponParams(),
-        Stripe.apiKey);
-    DeletedCoupon deletedCoupon = createdCoupon.delete(Stripe.apiKey);
-    assertTrue(deletedCoupon.getDeleted());
-    assertEquals(deletedCoupon.getId(), createdCoupon.getId());
-  }
+  public void testDelete() throws StripeException {
+    Coupon resource = Coupon.retrieve(RESOURCE_ID);
 
-  @Test
-  public void testCustomerCreateWithCouponPerCallAPIKey()
-      throws StripeException {
-    Coupon coupon = Coupon.create(getUniqueCouponParams(), Stripe.apiKey);
-    Map<String, Object> customerWithCouponParams = new HashMap<String, Object>();
-    customerWithCouponParams.put("coupon", coupon.getId());
-    Customer customer = Customer.create(customerWithCouponParams,
-        Stripe.apiKey);
-    assertEquals(customer.getDiscount().getCoupon().getId(), coupon.getId());
-
-    customer.deleteDiscount();
-    assertNull(Customer.retrieve(customer.getId()).getDiscount());
-  }
+    resource.delete();
```

Comment:
should we test `CouponDeleted` like you did for Card (if so, applies to all delete)

Reply:
Probably. The important part of the test is really `verifyRequest`, but might as well be consistent.